### PR TITLE
HOSTEDCP-1994: Add Secrets Store CSI RBAC Policies for HO and CPO

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
This PR adds the secrets store CSI driver to the go.mod and adds the needed Secrets Store CSI RBAC policies to the HO and CPO for ARO HCP deployments.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: support/azureutil/azureutil.go [hunk 1]
```diff
@@ -217,3 +218,8 @@ func VerifyResourceGroupLocationsMatch(ctx context.Context, hc *hyperv1.HostedCl
 
 	return nil
 }
+
+// IsAroHcp returns true if the managed service environment variable is set to ARO-HCP
+func IsAroHcp() bool {
```
Comment (csrwng) [inline support/azureutil/azureutil.go:223]: nit: s/Hcp/HCP/

File: cmd/install/assets/hypershift_operator.go [hunk 2]
```diff
@@ -941,7 +941,7 @@ type HyperShiftOperatorClusterRole struct {
 	EnableCVOManagementClusterMetricsAccess bool
 }
 
-func (o HyperShiftOperatorClusterRole) Build() *rbacv1.ClusterRole {
+func (o HyperShiftOperatorClusterRole) Build(managedService string) *rbacv1.ClusterRole {
```
Comment (csrwng) [inline cmd/install/assets/hypershift_operator.go:944]: This is not the pattern we've been using for the build functions. Generally any params are set as fields, the Build() has no args

**Review by csrwng — COMMENTED** [review]
A couple of comments

File: cmd/install/assets/hypershift_operator.go (hunk 2 above)
Comment (bryan-cox) [reply cmd/install/assets/hypershift_operator.go:944]: Ahh I see now what you mean.

**Pull request merged**